### ISSUE TITLE
Zen Mode Font Size

### ISSUE THREAD

**Issue by Jonatthu** [Microsoft/vscode#45617]
I request a special property for zen mode

zenMode.fontSize

So would be different from

editor.fontSize

**Comment by isidorn** [comment]
This seems like a special settings for zen mode which would take effect when in zen mode. Something like a language identifier settings which take priority when you are in javascript file for example.
```json
"[zen]": {
    "editor.fontSize": 12
}
```
Renaming this feature request to be more global.

**Comment by nealot** [comment]
It would be nice if this was applied more generally by allowing settings to be changed after a certain action has been executed.

**Comment by isidorn** [comment]
@sandy081 is this feature request realistic, could we add a new configuration scope for zen mode.
Here's a highly requested feature which would benefit from this https://github.com/Microsoft/vscode/issues/31609
We could also deprecat a lot of zenMode specific settings if we added this

**Comment by sandy081** [comment]
@isidorn Do you think all editor settings makes sense to be overridable in zen mode? This also brings up new overriding dimension, for eg zen mode takes precedence over language settings?

**Comment by isidorn** [comment]
@sandy081 yes I think all editor setting make sense to be overriden in zen mode. However in general I would need more than editor setting. Do for language overried you only allow editor ones to be overriden?
Yes, zen mode should take precedence over language settings imho.

In practice here are the most needed ones:
`editor.lineNumber`
`workbench.editor.showTabs`
`workbench.activityBar.visible`
`workbench.statusBar.visible`
`editor.fontSize`

**Comment by sandy081** [comment]
> Do for language overried you only allow editor ones to be overriden

Yes.

So you need support for more identifiers that can override settings (not just editor).

**Comment by isidorn** [comment]
@sandy081 yes. Does that overcomplicate things?

**Comment by sandy081** [comment]
Yes, it will for the user. It is hard to explain why some settings are supported for some override identifiers and not for others especially there is no UI. Also this will introduce two types of settings for zen mode, one those with `zen.*` and others with `[zen]`. Is not it simple if all settings follow only one type?

**Comment by isidorn** [comment]
@sandy081 once we introduce this I will deprecate all zen. settings

**Comment by sandy081** [comment]
Ok, I can provide following internally
- Contribute override identifiers with settings to override.

**Comment by Tobyee** [comment]
Hide minimap configurable +1

**Comment by Jonatthu** [comment]
@isidorn Is this going to be ever an option?

**Comment by Mattsteele1197** [comment]
I'd love to hide git changes and disable the minimap when I'm in Zen mode, but I need it in my normal workflow.

Has there been any updates to this, or workarounds with zenMode specific settings?

**Comment by hughlilly** [comment]
I’d like to be able to toggle `window.commandCenter` to off (`false`) when in Zen mode.

**Comment by miguno** [comment]
> I'd love to hide git changes and disable the minimap when I'm in Zen mode, but I need it in my normal workflow.
> 
> Has there been any updates to this, or workarounds with zenMode specific settings?

+1

**Comment by anatoly-yakubov-betpawa** [comment]
Centred mode has vertical rulers. 
1) Vertical rules (width) settings is being saved somewhere, but can only be configured by mouse. A numeric setting would be nice to have.
2) Would be nice to have zen.EditorMinWidth. Currently editor width is controlled by vertical rulers, which tied to % of width available, hence width changes on different displays and windowed/full-screen modes. New settings or two would give a better, more predictable control over focused editor size.

**Comment by nicksuhkeys** [comment]
any updates to this?

**Comment by faenuccio** [comment]
I would like to disable the double-click for MarkDown preview when in Zen mode: I often project MDs using ZenMode (and do not want double-click to trigger the window with the md original file), but modify them outside Zen (and then the double-click is useful).

**Comment by kkukshtel** [comment]
Would also love this. Zen-mode specific settings.

**Comment by talkohavy** [comment]
It's 2025, and still waiting for this feature 🙏🏼

**Comment by joeriddles** [comment]
Would love to be able to hide `editor.rulers` in zen mode

**Comment by dylan-k** [comment]
It would be great to have settings you could toggle when/if zen mode is active. This way, for example if you've got a typewriter-scrolling extension or similar, you could enable it during zen mode, but have it disabled otherwise.
